packages/algob/src/lib/tx.ts

```diff
@@ -70,15 +70,15 @@ export function makeAssetCreateTxn (
     flags.creator.addr,
     note,
     BigInt(asaDef.total),
-    asaDef.decimals,
-    asaDef.defaultFrozen,
-    asaDef.manager,
-    asaDef.reserve,
-    asaDef.freeze,
-    asaDef.clawback,
-    asaDef.unitName,
+    Number(asaDef.decimals),
+    asaDef.defaultFrozen ? asaDef.defaultFrozen : false,
+    asaDef.manager !== "" ? asaDef.manager : undefined,
+    asaDef.reserve !== "" ? asaDef.reserve : undefined,
+    asaDef.freeze !== "" ? asaDef.freeze : undefined,
+    asaDef.clawback !== "" ? asaDef.clawback : undefined,
+    asaDef.unitName ? asaDef.unitName : undefined,
```

Comment:
Can't we directly use `asaDef.unitName` here? 
Isn't `asaDef.unitName !== undefined ? asaDef.unitName : undefined` same as `asaDef.unitName` ? Or do we check for something else here (other than undefined)?

Reply:
yes